### PR TITLE
adds all test classes and directories

### PR DESCRIPTION
**Pull request by duane-mcmahon**
Adds test source directory and project structure mirroring src folder for unit testing. All concrete classes have had test classes auto-generated, etc, providing test skeleton classes for all relevant classes. Idea is to merge this on or by beginning of Iteration #6.

### CODE REVIEW

File: BCCCP.iml [hunk 1]
```diff
@@ -3,9 +3,20 @@
...
     <content url="file://$MODULE_DIR$">
+      <sourceFolder url="file://$MODULE_DIR$/test" isTestSource="true" />
       <sourceFolder url="file://$MODULE_DIR$/src" isTestSource="false" />
     </content>
     <orderEntry type="inheritedJdk" />
     <orderEntry type="sourceFolder" forTests="false" />
+    <orderEntry type="module-library" scope="TEST">
+      <library name="JUnit5">
+        <CLASSES>
+          <root url="jar://$MODULE_DIR$/lib/junit-jupiter-api-5.0.0-M5.jar!/" />
```
Comment (WBGroupR) [inline BCCCP.iml:14]: Jupiter seems to be where the respective imports are for JUnit5 - so just a query: this version of JUnit seems to be fairly recent release and not mentioned by Jim in his lecture/how-to. Is this likely to cause any issues with him I wonder - hopefully not!

**Review by WBGroupR — APPROVED** [review]
Thanks for setting this up Duane. Just have a couple of questions:

1) the version of JUnit (5) that this uses and whether Jim will find any issues with it?

2) We would also need to load the relevant Mockito libraries I assume (unless IntelliJ already has them)?

Can discuss above in team meeting today,
Wayne

File: BCCCP.iml (hunk 1 above)
Comment (duane-mcmahon) [reply BCCCP.iml:14]: it's largely the same as junit4 i believe, the differences relate to advanced features. i 've used junit5 for unit testing of all my assignments at csu this year without any issue.

Comment (duane-mcmahon) [reply BCCCP.iml:14]: actually this article summarises junit5 and the differences quite well: 
http://www.baeldung.com/junit-5

**Comment by WBGroupR** [comment]
Thanks Duane, that gives a good explanation, v5 looks fine to me.

Wayne

**Pull request merged**